## Q1. What are the three measures of central tendency?

    Mean, Median, Mode

## Q2. What is the difference between the mean, median, and mode? How are they used to measure the central tendency of a dataset?

The mean, median, and mode are measures of central tendency used to describe the central or typical value of a dataset.

Mean:
1. The mean, also known as the average, is calculated by summing up all the values in a dataset and dividing by the total number of observations.
2. Calculation: Mean = (Sum of all values) / (Total number of observations)
3. It is commonly used when the dataset follows a roughly symmetrical distribution and there are no extreme outliers.

Median:
1. The median is the middle value in a dataset when the observations are arranged in ascending or descending order.
2. Calculation: For an odd number of observations, the median is the middle value. For an even number of observations, the median is the average of the two middle values.
3. The median is often used when the data contains outliers or is skewed. It is less sensitive to extreme values compared to the mean. The median represents the value below which 50% of the observations fall.

Mode:
1. The mode is the value(s) that appear most frequently in a dataset.
2. The mode is determined by identifying the value(s) with the highest frequency.
3. The mode is useful for categorical data or when identifying the most common response or occurrence in a dataset. It can be used for both discrete and continuous variables.

## Q3. Measure the three measures of central tendency for the given height data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

1. Sum of all values = 2832.3

    Total number of observations = 16
    
    Mean = 2832.3/16 
    Mean = 177.01875

2. Median = As the dataset contains 16 observations, the middle value is the eighth value, which is 177.

   median = 177

3. Mode = the value with the highest frequency as the mode, the value 178 occurs three times

   Mode = 178

## Q4. Find the standard deviation for the given data:
[178,177,176,177,178.2,178,175,179,180,175,178.9,176.2,177,172.5,178,176.5]

In [1]:
import statistics

data = [178, 177, 176, 177, 178.2, 178, 175, 179, 180, 175, 178.9, 176.2, 177, 172.5, 178, 176.5]

# Calculate mean
mean = statistics.mean(data)

# Calculate median
median = statistics.median(data)

# Calculate mode
mode = statistics.mode(data)

# Calculate standard deviation
std_dev = statistics.stdev(data)

print("Mean:", mean)
print("Median:", median)
print("Mode:", mode)
print("Standard Deviation:", std_dev)

Mean: 177.01875
Median: 177.0
Mode: 178
Standard Deviation: 1.8472389305844188


## Q5. How are measures of dispersion such as range, variance, and standard deviation used to describe the spread of a dataset? Provide an example.

1. Measures of dispersion help to describe the variability in data. 
2. As the variability of the data increases the value of the measures of dispersion also increases.
3. Measures of dispersion include range, variance, standard deviation, mean deviation, quartile deviation, etc.

Range:
1. Range is the easiest measure of dispersion calculated by subtracting lowest value from maximum value.
2. The wide range indicates high variability and small range indicates low varibalilty in the distribution.
3. Rangeis very much affected by outliers.

Variance:
1. Variance indicates how far any number in a dataset from the mean.
2. Observations near to mean value get lowest result and far from mean gets higher value.
3. It is calculated by taking the average of the squared differences between each data point and the mean.

Standard deviation:
1. Standard deviation is square root of variance to get original value.
2. Low standard deviation indicates data point close to mean.

In [3]:
# example
data=[10,12,4,5,8,9,6,11,10]

data_range=max(data)-min(data)

mean = sum(data) / len(data)
variance = sum((x - mean) ** 2 for x in data) / len(data)

# Standard Deviation
std_deviation = variance **0.5

# Print the results
print("Range:", data_range)
print("Variance:", variance)
print("Standard Deviation:", std_deviation)

# The range provides a simple measure of the total spread,
# while the variance and standard deviation provide more precise information
# about the dispersion and variability in the dataset.

Range: 8
Variance: 6.888888888888889
Standard Deviation: 2.6246692913372702


## Q6. What is a Venn diagram?

Venn diagrams are commonly used to illustrate concepts related to set theory, logic, and statistics. They provide a clear and intuitive way to visualize the intersections and differences between different groups or categories of items.

## Q7. For the two given sets A = (2,3,4,5,6,7) & B = (0,2,6,8,10). Find:
(i) A B
(ii) A ⋃ B

In [4]:
A = {2, 3, 4, 5, 6, 7}
B = {0, 2, 6, 8, 10}

# Intersection of A and B
intersection = A.intersection(B)

# Union of A and B
union = A.union(B)

# Print the results
print("Intersection (A ∩ B):", intersection)
print("Union (A ⋃ B):", union)

Intersection (A ∩ B): {2, 6}
Union (A ⋃ B): {0, 2, 3, 4, 5, 6, 7, 8, 10}


## Q8. What do you understand about skewness in data?

1. Skewness is a statistical measure that helps us understand the lack of symmetry in the distribution of data.
2. It provides information about the shape of the distribution and the relative concentration of data points around the mean.
3. In symmetric distribution data points are evenly distributed around mean.
4. In skewed distribution data points tend to cluster more towards one tail of distribution.
5. skewness is classified into three categories - symmetric, positively and negatively skewed.
6. symmetric skewness - the data is symmetrically distributed around the mean, and relationship of mean, mode, median is equal
7. positive skewness - the tail on the right side of the distribution is longer or stretched out. The mean is typically greater than the median, and the majority of the data points are concentrated towards the lower values.
8. negative skewness - the tail on the left side of the distribution is longer or stretched out. The mean is typically less than the median, and the majority of the data points are concentrated towards the higher values.

## Q9. If a data is right skewed then what will be the position of median with respect to mean?

If a data distribution is right-skewed (positively skewed), the position of the median with respect to the mean will be to the left of the mean.

## Q10. Explain the difference between covariance and correlation. How are these measures used in statistical analysis?

Covariance and correlation are both measures used in statistical analysis to understand the relationship between two variables.

covariance :
1. Covariance measures the extent to which two variables vary together.
2. It quantifies the direction and strength of the linear relationship between two variables.
3. The covariance can be positive, indicating a positive relationship where both variables increase or decrease together, or negative, indicating an inverse relationship where one variable increases while the other decreases. 

correlation :
1. Correlation, on the other hand, measures both the direction and strength of the linear relationship between two variables. 2. It provides a standardized measure of the relationship that ranges from -1 to 1. 
3. A correlation of 1 indicates a perfect positive linear relationship, a correlation of -1 indicates a perfect negative linear relationship, and a correlation of 0 indicates no linear relationship. 
4. Correlation is unaffected by changes in scale or units of measurement, making it easier to compare and interpret.

Used :
1. Covariance and correlation can help in variable selection in regression analysis or other modeling techniques.
2. Covariance and correlation can be used to explore relationships in datasets, identify patterns, and gain insights into the data.

## Q11. What is the formula for calculating the sample mean? Provide an example calculation for a dataset.

Formula for sample mean: Sample Mean (x̄) = (Sum of all values in the dataset) / (Number of values in the dataset)

In [2]:
data = [10,22,11,25,15,26,16,18,20]

sum_num=sum(data)

den = len(data)

sample_mean= sum_num/den

print("Sum of all values in the dataset:",sum_num)
print("Number of values in the dataset:",den)
print("sample mean:",sample_mean)

Sum of all values in the dataset: 163
Number of values in the dataset: 9
sample mean: 18.11111111111111


## Q12. For a normal distribution data what is the relationship between its measure of central tendency?

1. The mean, median, and mode are all located at the center of the distribution.
2. The mean is equal to the median and mode in a symmetrical normal distribution.
3. The mean is the most commonly used measure of central tendency in a normal distribution.

## Q13. How is covariance different from correlation?

1. Covariance is primarily used to determine the presence and direction of a relationship between variables. However, it is not suitable for comparing the strength of relationships between different pairs of variables.
2. Correlation is widely used to assess the strength and direction of linear relationships between variables. It is particularly useful in comparing relationships and identifying the most influential factors in statistical analysis.

## Q14. How do outliers affect measures of central tendency and dispersion? Provide an example.

1. Outliers can have a significant impact on measures of central tendency and dispersion in a dataset.
2. Mean is sensitive to outliers, median and mode are less afected by outliers.
3. Range is greatly influenced by outlier as it consider minimum and maximum value.
4. variance and SD also sensitive to outliers

In [7]:
# example 
import numpy as np
from scipy import stats

# Dataset with an outlier
data=[10,20,15,10,25,30,26,18,13,22,200]

# Calculate mean
mean = np.mean(data)

# Calculate median
median = np.median(data)

# Calculate mode
mode = stats.mode(data)

# Calculate range
range_val = np.max(data) - np.min(data)

# Calculate variance
variance = np.var(data)

# Calculate standard deviation
std_dev = np.std(data)

# Output the results
print("Mean:", mean)
print("Median:", median)
print("Mode:", mode)
print("Range:", range_val)
print("Variance:", variance)
print("Standard Deviation:", std_dev)


Mean: 35.36363636363637
Median: 20.0
Mode: ModeResult(mode=array([10]), count=array([2]))
Range: 190
Variance: 2749.685950413223
Standard Deviation: 52.43744797769265


/tmp/ipykernel_70/1946096388.py:15: FutureWarning: Unlike other reduction functions (e.g. `skew`, `kurtosis`), the default behavior of `mode` typically preserves the axis it acts along. In SciPy 1.11.0, this behavior will change: the default value of `keepdims` will become False, the `axis` over which the statistic is taken will be eliminated, and the value None will no longer be accepted. Set `keepdims` to True or False to avoid this warning.
  mode = stats.mode(data)


1. mean value of 35.36, which is higher than the majority of the values in the dataset. 
2. the median remains unaffected at 20, providing a more robust measure of central tendency.
3. the range is greatly influenced by the outlier, resulting in a range of 190. 
4. the variance and standard deviation is also impacted by the outlier, yielding a higher value ,indicating a greater spread of values.